server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java

```diff
@@ -147,6 +147,7 @@ public MovAvgPipelineAggregationBuilder window(int window) {
         if (window <= 0) {
             throw new IllegalArgumentException("[window] must be a positive integer: [" + name + "]");
         }
+        HoltWintersModel.validateHWParams(model, window);
```

Comment:
Unless I'm missing something, all the validation in this class will be applied no matter the model used. Should we not be only applying the validation if it's holt-winters? Maybe we should have a (non-static) validate method on the model class and this class just calls validate on the model?

Reply:
The validate method makes sure the model is the correct type... but yeah it's sorta the wrong assignment of responsibilities (did it that way to minimize the number of `instanceof` checks scattered around).

++ to a validate method on the model itself, will move things around